01-JS-DOM-APIs/exerciseJS.js

```diff
@@ -0,0 +1,75 @@
+//EXERCISE 6
+  function eventRequest() {
+    var xhttp = new XMLHttpRequest();
+    xhttp.open('GET', 'http://api.icndb.com/jokes/random', true);
+    xhttp.onreadystatechange = function(event) {
+      var response = JSON.parse(event.target.response);
+      document.getElementById('sectionHidden').innerHTML = response.value.joke;
+       if (xhttp.readyState === XMLHttpRequest.DONE) {
+         if (xhttp.responseText.value) {
+             document.getElementById("buttonSection").innerHTML  = xhttp.responseText.value.joke;
+         }
+       }
+   }
+   xhttp.send();
+}
+//EXERCISE 7
+function eventOnClick(){
+  var config = {
+    url: 'http://api.icndb.com/jokes/random'
+};
+  function resolve(xhttp) {
+    console.log(xhttp.responseText);
+    var response = JSON.parse(event.target.response);
+    document.getElementById('sectionHidden').innerHTML = response.value.joke;
+};
+  function reject() {
+    alert('Something went wrong !');
+    document.getElementById("sectionHidden").style.color = "red";
+};
+  eventRequestReusable(config).then(resolve,reject);
+};
+//EXERCISE 9
+function secondEventOnClick(){
+  var config = {
+    url: 'https://api.github.com/search/repositories'
+};
+  function resolve(xhttp) {
+    console.log(xhttp.responseText);
+    var response = JSON.parse(event.target.response);
+    document.getElementById('sectionHidden').innerHTML = response.errors[0].field;
+};
+  function reject() {
+    alert('Something went wrong !');
+    document.getElementById("sectionHidden").style.color = "red";
+};
+  eventRequestReusable(config).then(resolve,reject);
+
+};
+//EXERCISE 7 AND 9
+function eventRequestReusable(object, resolve, reject) {
```

Comment:
I couldn't change name 'object' for 'config', the function stopped working when i did it.